MagenticOne Orchestrator Fixes

husseinmozannar opened the pull request:
## Why are these changes needed?

Add back some verification logic to MagenticOne Orchestrator.

Base the orchestrator on the base group chat manager.

Remaining:

- add intermediate logging
- update tests

## Related issue number


Will resolve #4177 


#3756  - Updates to this PR

husseinmozannar commented:
@ekzhu still working on this, but curious, is this better for integrating the M1 orchestrator as an actual group chat manager? Anything else that would improve it?

ekzhu reviewed (approved):
Looks good. Don't forget about adding unit tests for the magentic one group chat. Use ReplayChatCompletionClient #4412

ekzhu commented:
> @ekzhu still working on this, but curious, is this better for integrating the M1 orchestrator as an actual group chat manager? Anything else that would improve it?

Yes it is good to make it a group chat manager. It allows us to refactor group chat manager logic into a developer facing API.

ekzhu commented:
Also, remember to add API docs with examples and argument list for MagenticOneGroupChat

#4413

husseinmozannar commented:
Thanks Eric!

Added two simple tests and API docs, and ironed a few bugs + trace logger.

I ran into a mypy error because of overriding handle_agent_response and handle_start, added # type ignore because I couldn't figure out how to fix.

ekzhu commented:
One last small lint error see checks.

ekzhu commented:
> Thanks Eric!
> 
> 
> 
> Added two simple tests and API docs, and ironed a few bugs + trace logger.
> 
> 
> 
> I ran into a mypy error because of overriding handle_agent_response and handle_start, added # type ignore because I couldn't figure out how to fix.
> 
> 
> 
> Please merge when appropriate

Overriding handlers causes mypy error is a know issue.

ekzhu commented on python/packages/autogen-agentchat/src/autogen_agentchat/teams/_group_chat/_magentic_one/_magentic_one_group_chat.py:
```diff
@@ -13,18 +13,62 @@
+            from autogen_ext.models import OpenAIChatCompletionClient
+            from autogen_agentchat.agents import AssistantAgent
+            from autogen_agentchat.teams import MagenticOneGroupChat
+            from autogen_agentchat.task import TextMentionTermination, Console
```
```suggestion
            from autogen_agentchat.task import Console
```
